### PR TITLE
1.0.2

### PR DESCRIPTION
**Pull request by ernestmarcinko**
- Added an extra error check to WaifuRequestHandler
- Added a few more negative test assertions
- Code formatting
- License to composer.json

### CODE REVIEW

File: tests/WaifuTests/WaifuRequestHandlerTest.php [hunk 1]
```diff
@@ -154,6 +163,16 @@ public function testGetTrue(): void {
 		$response = $handler->make(RequestMethods::GET, 'fake')
 			->getTrue();
 		$this->assertSame(true, $response);
+
+		// trigger responseErrorCheck call without ->make()
+		try {
```
Comment (VictoriqueMoe) [inline tests/WaifuTests/WaifuRequestHandlerTest.php:168]: doesn't the php framework have some sort of `shouldThrow` assertion?

**Review by VictoriqueMoe — APPROVED** [review]

File: tests/WaifuTests/WaifuRequestHandlerTest.php (hunk 1 above)
Comment (ernestmarcinko) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: Yes, `$this->assertException(...)`

However the test can't continue afterwards, the function exits. I need to put each exception check to a separate function for that to work. This is the only way I could find to bypass that.

Comment (VictoriqueMoe) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: can you mock the exception and just assert the exception is the same as the mocked one? I expect it NOT to be able to continue, but that should be the last and only assertion in the test, no?

Comment (VictoriqueMoe) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: I am not fussed either way, just curious more than anything

Comment (ernestmarcinko) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: I don't know actually, let me check that. I think for that I would have to mock `throws ` keyword, I suspect it's not the exception the issue, but that it's thrown.

Comment (VictoriqueMoe) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: eh, i just assumed `assertException` would take an argument as to what the exception should be

Comment (ernestmarcinko) [reply tests/WaifuTests/WaifuRequestHandlerTest.php:168]: It does actually, but the code still stops, and still can't continue afterwards. I actually do that here: 
https://github.com/waifuvault/waifuVault-php-api/blob/e6dd34784fd525244c2d1df1d10917e334ed6b9d/tests/WaifuTests/WaifuRequestHandlerTest.php#L203-L208

But it's positioned as the last assertion in the function as it exits. All I could find is to do the try-catch that I did and compare the exception class + add an assert count to make sure it reaches the catch and compare at the end.

**Pull request merged**